lib/active_merchant/billing/gateway.rb

```diff
@@ -209,6 +209,20 @@ def add_field_to_post_if_present(post, options, field)
         post[field] = options[field] if options[field]
       end
 
+      def add_3ds_2_optional_fields(post, options)
+        three_ds_2 = options[:three_ds_2] || {}
+
+        if three_ds_2.has_key?(:optional)
+          three_ds_2[:optional].each do |key, value|
+            normalized_value = normalize(value)
+            next if normalized_value.nil?
```

Comment:
Could there be value in passing along `nil` values for some fields? I am thinking of APIs where some value has a default that you'd like to clear by sending `some_value: nil` or where the key is required but you have no value to pass along.

Reply:
it's better to not pass nil along. Since we're using normalize here, a '' or 'null' can turn into a nil.